sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -681,65 +681,76 @@ private void initialize() {
+                ManagedErrorLog errorLog = new ManagedErrorLog();
+                errorLog.setException(modelException);
+                errorLog.setTimestamp(new Date(minidumpDate));
+                errorLog.setFatal(true);
+                errorLog.setId(UUID.randomUUID());
+
+                /* Lookup app launch timestamp in session history. */
+                SessionContext.SessionInfo session = SessionContext.getInstance().getSessionAt(minidumpDate);
+                if (session != null && session.getAppLaunchTimestamp() <= minidumpDate) {
+                    errorLog.setAppLaunchTimestamp(new Date(session.getAppLaunchTimestamp()));
+                } else {
+
+                    /*
+                     * Fall back to log date if app launch timestamp information lost
+                     * or in the future compared to crash time.
+                     * This also covers the case where app launches then crashes within 1s:
+                     * app launch timestamp would have ms accuracy while minidump file is without
+                     * ms, in that case we also falls back to log timestamp
+                     * (this would be same result as truncating ms).
+                     */
+                    errorLog.setAppLaunchTimestamp(errorLog.getTimestamp());
+                }
 
                 /*
-                 * Fall back to log date if app launch timestamp information lost
-                 * or in the future compared to crash time.
-                 * This also covers the case where app launches then crashes within 1s:
-                 * app launch timestamp would have ms accuracy while minidump file is without
-                 * ms, in that case we also falls back to log timestamp
-                 * (this would be same result as truncating ms).
+                 * TODO The following properties are placeholders because fields are required.
+                 * They should be removed from schema as not used by server.
                  */
-                errorLog.setAppLaunchTimestamp(errorLog.getTimestamp());
-            }
+                errorLog.setProcessId(0);
+                errorLog.setProcessName("");
 
-            /*
-             * TODO The following properties are placeholders because fields are required.
-             * They should be removed from schema as not used by server.
-             */
-            errorLog.setProcessId(0);
-            errorLog.setProcessName("");
+                /*
+                 * TODO user id is read after restart contrary to Java crashes.
+                 * We should have a user history like we did for session to fix that issue.
+                 * The main issue with the current code is that userId can change between crash and reporting.
+                 */
+                errorLog.setUserId(UserIdContext.getInstance().getUserId());
+                try {
+                    if (savedDeviceInfo != null) {
```

Comment:
Where is the fallback to `getDeviceInfo(mContext)` if device file not found? We cannot rely on channel default behavior as the wrapper sdk name will be missing there and log will not be processed correctly if wrappersdk is not ndk.